### PR TITLE
StyleGAN

### PR DESCRIPTION
**Pull request by xdeng7**
Created stylegan to gluon-cv

### CODE REVIEW

File: scripts/gan/stylegan/demo_stylegan.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    generator = StyledGenerator(512)
+    generator.initialize()
+    generator.collect_params().reset_ctx(device)
+    generator.load_parameters(args.path, ctx=device)
+
+    mean_style = get_mean_style(generator, device)
+
+    step = int(math.log(args.size, 2)) - 2
+    
+    imgs = sample(generator, step, mean_style, args.n_sample, device)
```
Comment (bryanyzhu) [inline scripts/gan/stylegan/demo_stylegan.py:82]: Can the model be hybridized?

File: scripts/gan/stylegan/demo_stylegan.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        img_range = [min(data), max(data)]
+
+    norm_img = normalize_image(data, img_range[0], img_range[1])
+    img = nd.clip(norm_img * 255 + 0.5, 0, 255).asnumpy().astype(np.uint8) 
+    img = Image.fromarray(np.transpose(img, (1, 2, 0)))
+    img.save(file)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```
Comment (bryanyzhu) [inline scripts/gan/stylegan/demo_stylegan.py:59]: Can you write a README for this folder, like cycleGAN folder did? For example, how to prepare the training data, how to train and how to evaluate the model, etc. Thank you.

Comment (xdeng7) [reply scripts/gan/stylegan/demo_stylegan.py:59]: > Can you write a README for this folder, like cycleGAN folder did? For example, how to prepare the training data, how to train and how to evaluate the model, etc. Thank you.

Sure, no problem.

Comment (xdeng7) [reply scripts/gan/stylegan/demo_stylegan.py:59]: I've added README. Please check.

**Comment by bryanyzhu** [comment]
Hi @zhreshold Any idea on why this error happens for py2 test? It starts to show up since I merged this PR #1388 . Thank you.

```
File "/var/lib/jenkins/miniconda3/envs/gluon_cv_py2_test/lib/python2.7/site-packages/portalocker/utils.py", line 290

    def __init__(self, maximum: int, name: str = 'bounded_semaphore',

                              ^

SyntaxError: invalid syntax
```

**Comment by zhreshold** [comment]
@bryanyzhu type annotation is not supported in python2, we can now
-  fix it
- or officially drop python2 support and turn off the python2 unittests in CI

**Comment by bryanyzhu** [comment]
@xdeng7 Can you rebase with master and push again? We have dropped unittest for python2, so the original error shouldn't be there anymore.

**Comment by xdeng7** [comment]
> @xdeng7 Can you rebase with master and push again? We have dropped unittest for python2, so the original error shouldn't be there anymore.

Thanks for helping out. I will try it.

**Comment by mli** [comment]
Job PR-1395-13 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1395/13/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/gan/stylegan/model.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+from math import sqrt
+
+import mxnet as mx
+import mxnet.ndarray as nd
+
+from modules import *
+# pylint: disable-all
+
+class Generator(nn.HybridBlock):
+    def __init__(self, code_dim, fused=True):
```
[the remark this answers is not in the document]
Comment (xdeng7) [reply scripts/gan/stylegan/model.py]: Fixed! Please check.

**Comment by mli** [comment]
Job PR-1395-14 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1395/14/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1395-15 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1395/15/index.html 
Code coverage of this PR:  vs. Master:

**Comment by bryanyzhu** [comment]
LGTM, thank you for your contribution, will merge it now.

**Pull request merged**